xdsl/traits.py

```diff
@@ -342,3 +344,18 @@ def get_callable_region(cls, op: Operation) -> Region:
         Returns the body of the operation
         """
         raise NotImplementedError
+
+
+@dataclass(frozen=True)
+class HasCanonicalisationPatternsTrait(OpTrait):
+    """
+    Provides the rewrite passes to canonicalize an operation.
```

Comment:
Can you add in the documentation that the rewrite pattern should match the given operation type as root?